soda/core/tests/data_source/test_profile_columns.py

```diff
@@ -156,113 +169,218 @@ def test_profile_columns_all_tables_all_columns(data_source_fixture: DataSourceF
+            {},
+            id="ensure 'like' operator is not used on columns without presence of wildcard",
+        ),
+        pytest.param(
+            """
+                profile columns:
+                    columns:
+                        - include %.ITEMS_SOLD
+                        - exclude %.%ITEMS
+            """,
+            {
+                "table_name1": ["ITEMS_SOLD"],
+            },
+            id="ensure matching pattern doesn't just consider start of table/column name",
         ),
     ],
 )
 def test_profile_columns_inclusions_exclusions(
-    data_source_fixture: DataSourceFixture, table_name, soda_cl_str, expectation
+    data_source_fixture: DataSourceFixture, soda_cl_str, expected_column_profiling_results
 ):
-    _table_name = data_source_fixture.ensure_test_table(table_name)
+    _table_name1 = data_source_fixture.ensure_test_table(customers_profiling)
+    _table_name2 = data_source_fixture.ensure_test_table(customers_profiling_capitalized)
     scan = data_source_fixture.create_test_scan()
     mock_soda_cloud = scan.enable_mock_soda_cloud()
-    scan.add_sodacl_yaml_str(soda_cl_str.format(table_name=_table_name))
+    scan.add_sodacl_yaml_str(soda_cl_str.format(table_name1=_table_name1, table_name2=_table_name2))
     # TODO: we should only allow warnings here, we'll have to look at what the errors were
     # it is most likely will be related to https://sodadata.atlassian.net/browse/CLOUD-155
     scan.execute(allow_error_warning=True)
-    profiling_result = mock_soda_cloud.pop_scan_result()
-    for table_result in profiling_result["profiling"]:
-        if table_result["table"].lower().startswith("sodatest_customer"):
-            column_names = [col_profile["columnName"] for col_profile in table_result["columnProfiles"]]
-            if expectation == "all but id":
-                assert "id" not in column_names
-            elif expectation == "all but nothing":
-                assert len(column_names) == 0
-            else:
-                assert "id" not in column_names
-                assert "cst_size" in column_names
-                assert "country" not in column_names
+    scan_results = mock_soda_cloud.pop_scan_result()
+
+    profiled_tables = [profiled_table for profiled_table in scan_results["profiling"]]
+    profiled_tables = sorted(profiled_tables, key=lambda x: x["table"].lower())
+
+    column_profiling_results = {
+        f"table_name{index}": list(map(lambda x: x["columnName"], profiled_table["columnProfiles"]))
+        for index, profiled_table in enumerate(profiled_tables, 1)
+    }
+
+    test_data_source = os.environ.get("test_data_source")
+
+    test_data_source_uppercase = test_data_source in UPPERCASE_COLUMN_NAME_DATABASES
+    test_data_source_lowercase = test_data_source in LOWERCASE_COLUMN_NAME_DATABASES
+
+    if test_data_source_uppercase or test_data_source_lowercase:
+        casify_function = lambda x: x.upper() if test_data_source_uppercase else x.lower()
```

Comment:
why we don't use default_casify_function in data_source.py? Can we handle the casing in soda-core instead of tests?